src/Servers/Kestrel/samples/WebTransportSampleApp/WebTransportSampleApp/Program.cs

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Net.Security;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Internal;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.Https;
+
+namespace WebTransportSample;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var hostBuilder = new HostBuilder()
+            .ConfigureLogging((_, factory) =>
+            {
+                factory.SetMinimumLevel(LogLevel.Trace);
+                factory.AddSimpleConsole(o => o.TimestampFormat = "[HH:mm:ss.fff] ");
+            })
+            .ConfigureWebHost(webHost =>
+            {
+                var cert = CertificateLoader.LoadFromStoreCert("localhost", StoreName.My.ToString(), StoreLocation.CurrentUser, false);
+
+                webHost.UseKestrel()
+                .ConfigureKestrel((context, options) =>
+                {
+                    // Port configured for WebTransport
+                    options.Listen(IPAddress.Any, 5007, listenOptions =>
+                    {
+                        listenOptions.UseHttps(GenerateManualCertificate());
+                        listenOptions.UseConnectionLogging();
+                        listenOptions.Protocols = HttpProtocols.Http1AndHttp2AndHttp3;
+                    });
+                })
+                .UseStartup<Startup>();
+            });
+
+        var host = hostBuilder.Build();
+
+        // Listener needs to be configured before host (and HTTP/3 endpoints) start up.
+        using var httpEventSource = new HttpEventSourceListener(host.Services.GetRequiredService<ILoggerFactory>());
+
+        host.Run();
+    }
+
+    // Adapted from: https://github.com/wegylexy/webtransport
+    // We will need to eventually merge this with existing Kestrel certificate generation
+    // tracked in issue #41762
+    private static X509Certificate2 GenerateManualCertificate()
```

Comment:
Is it ok to leave this? I am just unsure how else to do it such that the application can output the hash so the JS can use it?

Reply:
This is fine